jobsdb/unionQuery.go

```diff
@@ -18,8 +18,26 @@ type MultiTenantHandleT struct {
 	*HandleT
 }
 
+type workspacePickup struct {
+	Limit        int
+	PayloadLimit int64
+	AfterJobID   *int64
+}
+
+type (
+	MoreToken interface{}
+	moreToken struct {
+		afterJobIDs map[string]*int64
```

Comment:
**Explain:** for fair pickup algorithm we need to keep track of all the max job IDs per workspace